code/extensions/DMSSiteTreeExtension.php

```diff
@@ -52,16 +52,6 @@ public function updateCMSFields(FieldList $fields)
             ));
     }
 
-    /**
-     * Get a list of document sets for the owner page
-     *
-     * @return ArrayList
-     */
-    public function getDocumentSets()
-    {
-        return $this->owner->DocumentSets();
```

Comment:
I'm fairly certain that the 500 error will be coming from a circular reference in this method, which has been appropriately removed in this PR.
However in order to achieve the same thing without the alteration of the public API (i.e. adhere to semver patch level) I think simply changing this one line would be sufficient.
Utilising [`hasManyComponent`](https://github.com/silverstripe/silverstripe-framework/blob/3.6/model/DataObject.php#L2003) this would become
`return $this->owner->hasManyComponent('DocumentSets');`
or something to that affect.

Beyond that I see that `master` only has bugfixes in it, so there's no other reason why this couldn't go into a patch release for `2.1`.

Reply:
That's correct, this works. But I thought it would be cleaner if we removed the method as it doesnt provide any additional benefit. Happy to revert if you want me to?

Reply:
True, can also deprecate, it would be good. Would require a minor version release, but I'm OK with that.